### PR TITLE
[RUNTIME] Change state of runtime based on Dawn message

### PR DESCRIPTION
**Pull request by brandonxxlee**
Dawn sends via UDP the current state, and now runtime changes it state correctly
to reflect changes in the message Dawn sends.

This PR is waiting on #150 to be merged first.

### CODE REVIEW

File: runtime/testy/runtime.py [hunk 1]
```diff
@@ -83,11 +83,14 @@ def nonTestModePrint(*args):
...
-          spawnProcess(PROCESS_NAMES.STUDENT_CODE, runStudentCode, testName, maxIter)
+          name = "teleop"
+          if testName:
+              name = testName
+          spawnProcess(PROCESS_NAMES.STUDENT_CODE, runStudentCode, name, maxIter)
           controlState = "teleop"
           continue
         elif newBadThing.event == BAD_EVENTS.ENTER_AUTO and controlState != "auto":
-          # spawnProcess(autonomous code)
+          spawnProcess(PROCESS_NAMES.STUDENT_CODE, runStudentCode, "autonomous")
```
Comment (nikitakit) [inline runtime/testy/runtime.py]: Based on the changes here, I still don't see whether #95 has been addressed. Could you comment on that?

**Review by nikitakit — COMMENTED** [review]
I agree to wait for #150 to be merged first

**Comment by nikitakit** [comment]
lgtm now, but waiting on #150

File: runtime/testy/runtime.py [hunk 2]
```diff
@@ -83,11 +83,16 @@ def nonTestModePrint(*args):
           dawn_connected = False
           continue
         elif newBadThing.event == BAD_EVENTS.ENTER_TELEOP and controlState != "teleop":
-          spawnProcess(PROCESS_NAMES.STUDENT_CODE, runStudentCode, testName, maxIter)
+          terminate_process(PROCESS_NAMES.STUDENT_CODE)
+          name = "teleop"
+          if testName:
```
Comment (yizow) [inline runtime/testy/runtime.py:88]: i personally prefer the `if ... else ...` for clarity, but that may just be personal preference.

**Review by yizow — COMMENTED** [review]
LGTM

**Review by nikitakit — APPROVED** [review]
lgtm

**Pull request merged**